packages/accounts/src/kernel-zerodev/provider.ts

```diff
@@ -117,43 +125,101 @@ export class ZeroDevProvider extends SmartAccountProvider<HttpTransport> {
-      this.customMiddleware ?? noOpMiddleware
-    )({
-      initCode,
-      sender: this.getAddress(),
-      nonce: this.account.getNonce(),
-      callData,
-      signature: this.account.getDummySignature(),
-    } as UserOperationStruct);
-
-    const request = deepHexlify(await resolveProperties(uoStruct));
-    if (!isValidRequest(request)) {
-      // this pretty prints the uo
-      throw new Error(
-        `Request is missing parameters. All properties on UserOperationStruct must be set. uo: ${JSON.stringify(
-          request,
-          null,
-          2
-        )}`
-      );
-    }
+    let hash: string = "";
+    let request: UserOperationRequest;
+    let i = 0;
+    let maxFeePerGas = 0n;
+    let maxPriorityFeePerGas = 0n;
+    do {
+      const uoStruct = await asyncPipe(
+        this.dummyPaymasterDataMiddleware,
+        this.feeDataGetter,
+        this.paymasterDataMiddleware,
+        this.gasEstimator,
+        this.customMiddleware ?? noOpMiddleware
+      )({
+        initCode,
+        sender: this.getAddress(),
+        nonce: this.account.getNonce(),
+        callData,
+        signature: this.account.getDummySignature(),
+        maxFeePerGas,
+        maxPriorityFeePerGas,
+      } as UserOperationStruct);
+
+      request = deepHexlify(await resolveProperties(uoStruct));
+      if (!isValidRequest(request)) {
+        // this pretty prints the uo
+        throw new Error(
+          `Request is missing parameters. All properties on UserOperationStruct must be set. uo: ${JSON.stringify(
+            request,
+            null,
+            2
+          )}`
+        );
+      }
+
+      request.signature = await this.account.validator.getSignature(request);
 
-    request.signature = await this.account.validator.getSignature(request);
+      try {
+        hash = await this.rpcClient.sendUserOperation(
+          request,
+          this.entryPointAddress
+        );
+      } catch (error: any) {
+        if (this.isReplacementOpError(error) && i++ < this.sendTxMaxRetries) {
+          maxFeePerGas = (BigInt(request.maxFeePerGas) * 113n) / 100n;
+          maxPriorityFeePerGas =
+            (BigInt(request.maxPriorityFeePerGas) * 113n) / 100n;
+          console.log(
+            `Resending tx with Increased Gas fees: maxFeePerGas: ${maxFeePerGas}, maxPriorityFeePerGas: ${maxPriorityFeePerGas}`
+          );
+          continue;
+        }
+        throw this.unwrapError(error);
+      }
+    } while (hash === "");
 
     return {
-      hash: await this.rpcClient.sendUserOperation(
-        request,
-        this.entryPointAddress
-      ),
+      hash,
       request,
     };
   };
 
+  isReplacementOpError(errorIn: any): boolean {
+    if (errorIn.cause != null) {
+      const failedOpMessage: string | undefined = errorIn?.cause?.message;
+      return (
+        failedOpMessage?.includes(
+          "replacement op must increase maxFeePerGas and MaxPriorityFeePerGas"
+        ) ?? false
```

Comment:
Alchemy's Bundler and Stackup's Bundler have, unfortunately, different message. We should add a comment here to say we need to also need to check for Alchemy's message

Reply:
Oh makes sense! I can't seem to find the code for alchemy's bundler. And it's bit hard to reproduce the error.
According to their docs, it seems Alchemy's bundler throws `Replacement Underpriced Error` error message. [Doc Link](https://docs.alchemy.com/reference/eth-senduseroperation#replacement-underpriced-error)

I will try to reproduce it somehow, confirm and include the logic.

Reply:
Found the error log from discord: https://discord.com/channels/735965332958871634/735965333600338072/1127962136962932818